core/src/main/java/org/apache/spark/util/collection/TimSort.java

```diff
@@ -914,7 +915,7 @@ private void mergeHi(int base1, int len1, int base2, int len2) {
      * @param minCapacity the minimum required capacity of the tmp array
      * @return tmp, whether or not it grew
      */
-    private Buffer ensureCapacity(int minCapacity) {
+    private Buffer ensureCapacity(int minCapacity) throws IOException {
```

Comment:
I use `IOException` because in some existing methods that do similar thing `IOException` is thrown instead of `OutOfMemoryError`. E.g., `UnsafeExternalSorter`.